### PR TITLE
fix(Core/Players): Remove shapeshift auras on taxi restore.

### LINKED ISSUE #7240: [Druid] Shapeshift remains after leaving RDF and on a flight
### What client do you play on?

enUS

### Faction

- [X] Alliance
- [X] Horde

### Content Phase:

- [X] Generic
- [ ] 1-19
- [ ] 20-29
- [ ] 30-39
- [ ] 40-49
- [ ] 50-59

### Current Behaviour

Original report: [Chromie issue#1329](https://github.com/chromiecraft/chromiecraft/issues/1329)

If you're on a a flight when the RDF queue pops, then when you come out (if you were in a form when you leave the dungeon) you'll be **still** in form outside... on the flight...

### Expected Blizzlike Behaviour

Revert to caster form when on a flight.

### Source

WoWScrnShot_080221_204636

### Steps to reproduce the problem

1. Take a Flight Point somewhere.
2. Have RDF pop while on the flight.
3. Leave RDF while in a druid form.

### Extra Notes

_No response_

### AC rev. hash/commit

https://github.com/chromiecraft/azerothcore-wotlk/commit/3d4befd7043fbbceb32f7c2a6e0a4f9f638f8b0a

### Operating system

Ubuntu 20.04

### Modules

- [mod-ah-bot](https://github.com/azerothcore/mod-ah-bot)
- [mod-cfbg](https://github.com/azerothcore/mod-cfbg)
- [mod-chromie-xp](https://github.com/azerothcore/mod-chromie-xp)
- [mod-desertion-warnings](https://github.com/azerothcore/mod-desertion-warnings)
- [mod-duel-reset](https://github.com/azerothcore/mod-duel-reset)
- [mod-eluna-lua-engine](https://github.com/azerothcore/mod-eluna-lua-engine)
- [mod-ip-tracker](https://github.com/azerothcore/mod-ip-tracker)
- [mod-low-level-arena](https://github.com/azerothcore/mod-low-level-arena)
- [mod-multi-client-check](https://github.com/azerothcore/mod-multi-client-check)
- [mod-pvp-titles](https://github.com/azerothcore/mod-pvp-titles)
- [mod-pvpstats-announcer](https://github.com/azerothcore/mod-pvpstats-announcer)
- [mod-queue-list-cache](https://github.com/azerothcore/mod-queue-list-cache)
- [mod-server-auto-shutdown](https://github.com/azerothcore/mod-server-auto-shutdown)
- [lua-CarbonCopy](https://github.com/55Honey/Acore_CarbonCopy)
- [lua-CustomWorldboss](https://github.com/55Honey/Acore_CustomWorldboss)
- [lua-LevelUpReward](https://github.com/55Honey/Acore_LevelUpReward)
- [lua-send-and-bind](https://github.com/55Honey/Acore_SendAndBind)
- [lua-Zonecheck](https://github.com/55Honey/acore_Zonecheck)

### Customizations

None

### Server

ChromieCraft

Confirmed on local AC (15013d2):
If you enter a dungeon through the dungeon finder as a druid, and you are already on a (gryphon) flight, you will keep your shapeshifted form, that you had, when you left the dungeon group or teleported out of the dungeon:
unknown_2021 08 05-14 10

### PR DESCRIPTION
**Pull request by UltraNix**
Fixed #7240

## Changes Proposed:
-  Remove shapeshift auras on taxi restore when leaving bg/rdf.

## Issues Addressed:

- Closes #7240
- Closes https://github.com/chromiecraft/chromiecraft/issues/1329

## Tests Performed:

- Not tested

## How to Test the Changes:

1. Take a Flight Point somewhere.
2. Have RDF pop while on the flight.
3. Leave RDF while in a druid form.

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: src/server/game/Entities/Player/Player.cpp [hunk 1]
```diff
@@ -9723,6 +9723,9 @@ void Player::ContinueTaxiFlight()
         return;
     }
 
+    if (IsInDisallowedMountForm())
+        RemoveAurasByType(SPELL_AURA_MOD_SHAPESHIFT);
```
Comment (Helias) [inline src/server/game/Entities/Player/Player.cpp:9727]:
```suggestion
    if (IsInDisallowedMountForm())
    {
        RemoveAurasByType(SPELL_AURA_MOD_SHAPESHIFT);
    }
```

**Review by Helias — APPROVED** [review]

**Comment by Helias** [comment]
it seems more an hackfix, we should remove all the auras similar to shapeshift

**Comment by pklloveyou** [comment]
image
Failed the test

**Comment by pklloveyou** [comment]
Tested

**Pull request merged**